Runtime/Services/BacktraceDatabaseContext.cs

```diff
@@ -302,19 +222,15 @@ private void RemoveMaxRetries()
             for (int i = 0; i < total; i++)
             {
                 var value = currentBatch[i];
-                if (value.Valid())
+                if (value.Count > 0)
                 {
-                    value.Delete();
-                    if (value.Count > 0)
-                    {
-                        TotalRecords = TotalRecords - value.Count;
-                    }
-                    else
-                    {
-                        TotalRecords--;
-                    }
-                    TotalSize -= value.Size;
+                    TotalRecords = TotalRecords - value.Count;
+                }
+                else
+                {
+                    TotalRecords--;
                 }
+                TotalSize -= value.Size;
```

Comment:
Do we need to remove this key once we're done like we do in the delete function?

Reply:
Here we're decreasing the size of the database directory. To avoid checking database size every time, we're monitoring it here.